File: RobWork/example/java/src/ExFindFromWorkCell.java
import org.robwork.sdurw.*;

public class ExFindFromWorkCell {
    public static void findFromWorkCell(WorkCellPtr wc)
    {
        // get the default state
        State state = wc.getDefaultState();
        Frame worldFrame = wc.getWorldFrame();
        // find a frame by name, remember NULL is a valid return
        Frame frame = wc.findFrame("FixedFrameName");
        // find a frame by name, but with a specific frame type
        FixedFrame fframe = wc.findFixedFrame("FixedFrameName");
        MovableFrame mframe = wc.findMovableFrame("MovableFrameName");
        // find a device by name
        DevicePtr device = wc.findDevice("SerialDeviceName");
        SerialDevicePtr sdevice = wc.findSerialDevice("SerialDeviceName");
    }
}

File: RobWork/example/python/ex_find_from_workcell.py
from sdurw import *

def findFromWorkCell(wc):
    # get the default state
    state = wc.getDefaultState();
    worldFrame = wc.getWorldFrame();
    # find a frame by name, remember NULL is a valid return
    frame = wc.findFrame("FixedFrameName");
    # find a frame by name, but with a specific frame type
    fframe = wc.findFixedFrame("FixedFrameName");
    mframe = wc.findMovableFrame("MovableFrameName");
    # find a device by name
    device = wc.findDevice("SerialDeviceName");
    sdevice = wc.findSerialDevice("SerialDeviceName");